examples/basic/src/VideoPlayer.tsx

```diff
@@ -25,53 +25,95 @@ import Video, {
   BufferingStrategyType,
   ReactVideoSource,
   SelectedTrackType,
+  ResizeMode,
+  VideoTrack,
+  SelectedTrack,
+  VideoDecoderProperties,
+  SelectedVideoTrack,
 } from 'react-native-video';
 import styles from './styles';
 import {AdditionalSourceInfo} from './types';
 import {
   bufferConfig,
-  defaultValue,
+  isAndroid,
+  isIos,
+  samplePoster,
   srcList,
   textTracksSelectionBy,
 } from './constants';
-import {Overlay, toast} from './components';
+import {
+  AudioTrackSelector,
+  Indicator,
+  Seeker,
+  TextTrackSelector,
+  toast,
+  TopControl,
+  VideoTrackSelector,
+} from './components';
+import ToggleControl from './ToggleControl.tsx';
+import MultiValueControl, {
+  MultiValueControlPropType,
+} from './MultiValueControl.tsx';
 
 type Props = NonNullable<unknown>;
 
 const VideoPlayer: FC<Props> = ({}) => {
-  const [state, setState] = useState(defaultValue);
+  const [rate, setRate] = useState(1);
+  const [volume, setVolume] = useState(1);
+  const [muted, setMuted] = useState(false);
+  const [resizeMode, setResizeMode] = useState(ResizeMode.CONTAIN);
+  const [duration, setDuration] = useState(0);
+  const [currentTime, setCurrentTime] = useState(0);
+  const [_, setVideoSize] = useState({videoWidth: 0, videoHeight: 0});
+  const [paused, setPaused] = useState(false);
+  const [fullscreen, setFullscreen] = useState(true);
+  const [decoration, setDecoration] = useState(true);
+  const [isLoading, setIsLoading] = useState(false);
+  const [audioTracks, setAudioTracks] = useState<AudioTrack[]>([]);
+  const [textTracks, setTextTracks] = useState<TextTrack[]>([]);
+  const [videoTracks, setVideoTracks] = useState<VideoTrack[]>([]);
+  const [selectedAudioTrack, setSelectedAudioTrack] = useState<
+    SelectedTrack | undefined
+  >(undefined);
+  const [selectedTextTrack, setSelectedTextTrack] = useState<
+    SelectedTrack | undefined
+  >(undefined);
+  const [selectedVideoTrack, setSelectedVideoTrack] =
+    useState<SelectedVideoTrack>({
+      type: SelectedVideoTrackType.AUTO,
+    });
+  const [srcListId, setSrcListId] = useState(0);
+  const [repeat, setRepeat] = useState(false);
+  const [controls, setControls] = useState(false);
+  const [useCache, setUseCache] = useState(false);
+  const [poster, setPoster] = useState<string | undefined>(undefined);
+  const [showNotificationControls, setShowNotificationControls] =
+    useState(false);
+  const [isSeeking, setIsSeeking] = useState(false);
```

Comment:
@seyedmostafahasani it's not necessary but how about to move it to `useReducer`? I think It would be more readable (btw sorry for long review time)

Reply:
@KrzysztofMoch Actually, I don't particularly like `useReducer` 😉, but if you want, I can use it.

Reply:
The best solution is to create a hook in RNV to retrieve all these informations from the player !

Reply:
@freeboub as @KrzysztofMoch said earlier, I will handle it with `useReducer`.

Reply:
I don't think this is a good solution ... 
You will introduce new dependency in the sample, I don't think this is the way to go as all app have similar issues. It would be better to provide a common solution for that !
Please don't do it in this PR